### ISSUE TITLE
new method for verifying git is not working

### ISSUE THREAD

**Issue by hagaivcita** [rdp/specific_install#54]
the new method for checking if `git` binaries are installed is not working in every situations since ruby's `system` call does not find shell builtins (ruby 2.7.0).

while using this Gem in my docker environment using `FROM ubuntu:18.04` and `ruby 2.7.0` the invocation of `system("type -a git")` returns `nil`, specifying the command `type` was not found even though it is a shell builtin.

**Comment by zph** [comment]
Interesting.

Can you please check what shell it's using here? It's possible we need a better portable check for git being on path.

I'm assuming shell is sh in this situation.

**Comment by zph** [comment]
Can you please provide the minimal docker file that you're using and the commands to replicate this?

**Comment by dabits** [comment]
I have same problem on ruby 2.6.5.

Here is commands:
```
$ docker pull ruby:2.6.5-alpine3.9
$ docker run --rm -it ruby:2.6.5-alpine3.9 sh
/ # irb
irb(main):001:0> system("type -a git")
=> nil
irb(main):002:0> system("which git") || system("where git")
/usr/bin/git
=> true
```

**Comment by djpate** [comment]
Same thing for us. Ruby 3 alpine

**Comment by liamdawson** [comment]
I think it's because `type` (and `command`) are both shell builtins on that image:

```shell
$ docker run --rm -it ruby:2.6.5-alpine3.9 sh
/ # type which
which is /usr/bin/which
/ # type type
type is a shell builtin
/ # type command
command is a shell builtin
/ # irb
irb(main):001:0> system("which sh")
/bin/sh
=> true
irb(main):002:0> system("type -a sh")
=> nil
irb(main):003:0> system("sh -c 'type -a sh'")
/bin/sh
=> true
```

I would have suggested `command -v`, but command is a builtin as well.

**Comment by rdp** [comment]
Please try 0.3.7, thanks for the report and fix!

**Comment by hagaivcita** [comment]
fixed by #55
